MapboxNavigation/Abbreviations.swift

```diff
@@ -0,0 +1,52 @@
+import UIKit
+
+let path = Bundle.navigationUI.path(forResource: "Abbreviations", ofType: "plist")!
+let allAbbrevations = NSDictionary(contentsOfFile: path) as? [String: [String: String]]
+
+/// Options that specify what kinds of words in a string should be abbreviated.
+struct StringAbbreviationOptions : OptionSet {
+    let rawValue: Int
+    
+    /// Abbreviates ordinary words that have common abbreviations.
+    static let Abbreviations = StringAbbreviationOptions(rawValue: 1 << 0)
+    /// Abbreviates directional words.
+    static let Directions = StringAbbreviationOptions(rawValue: 1 << 1)
+    /// Abbreviates road name suffixes.
+    static let Classifications = StringAbbreviationOptions(rawValue: 1 << 2)
+}
+
+extension String {
+    /// Returns an abbreviated copy of the string.
+    func abbreviated(by options: StringAbbreviationOptions) -> String {
+        return characters.split(separator: " ").map(String.init).map { (word) -> String in
+            let lowercaseWord = word.lowercased()
+            if let abbreviation = allAbbrevations!["abbreviations"]![lowercaseWord], options.contains(.Abbreviations) {
+                return abbreviation
+            }
+            if let direction = allAbbrevations!["directions"]![lowercaseWord], options.contains(.Directions) {
+                return direction
+            }
+            if let classification = allAbbrevations!["classifications"]![lowercaseWord], options.contains(.Classifications) {
```

Comment:
It might be cleaner to define a few more constants up top so you can access dictionaries like `classifications` directly.